test/lib/rubycritic/revision_comparator_test.rb

```diff
@@ -0,0 +1,63 @@
+require 'test_helper'
+require 'rubycritic/revision_comparator'
+
+describe Rubycritic::RevisionComparator do
+  subject { Rubycritic::RevisionComparator.new([]) }
+
+  describe '#set_statuses' do
+    context 'in a SCS with :revision? == false' do
+      before do
+        Rubycritic::Config.expects(:source_control_system)
+                          .at_least_once
+                          .returns(stub(revision?: false))
+      end
+
+      it 'does not attempt to compare with previous results' do
+        subject.expects(:load_cached_analysed_modules).never
+        result = subject.set_statuses([])
+        result.must_equal([])
+      end
+    end
+
+    context 'in a SCS with :revision? == true' do
+      before do
+        Rubycritic::Config.expects(:source_control_system)
+                          .at_least_once
+                          .returns(stub(revision?: true))
+      end
+
+      context 'without previously cached results' do
+        before do
+          subject.expects(:revision_file).returns('foo')
+          File.expects(:file?).with('foo').returns(false)
+        end
+
+        it 'attempts to load previously cached results' do
```

Comment:
Yeah, probably just need to test that it does not invoke SmellsStatusSetter.

I'll fix it once I get back to my pc